integration/container/create_test.go

```diff
@@ -301,3 +303,83 @@ func TestCreateWithCustomReadonlyPaths(t *testing.T) {
 		checkInspect(t, ctx, name, tc.expected)
 	}
 }
+
+func TestCreateWithInvalidHealthcheckParams(t *testing.T) {
+	defer setupTest(t)()
+
+	testCases := []struct {
+		doc         string
+		interval    time.Duration
+		timeout     time.Duration
+		retries     int
+		startPeriod time.Duration
+		expectedErr string
+	}{
+		{
+			doc:         "test invalid Interval in Healthcheck: less than 0s",
+			interval:    -10 * time.Millisecond,
+			timeout:     time.Second,
+			retries:     1000,
+			expectedErr: fmt.Sprintf("Interval in Healthcheck cannot be less than %s", container.MinimumDuration),
+		},
+		{
+			doc:         "test invalid Interval in Healthcheck: larger than 0s but less than 1ms",
+			interval:    500 * time.Microsecond,
+			timeout:     time.Second,
+			retries:     1000,
+			expectedErr: fmt.Sprintf("Interval in Healthcheck cannot be less than %s", container.MinimumDuration),
+		},
+		{
+			doc:         "test invalid Timeout in Healthcheck: less than 1ms",
+			interval:    time.Second,
+			timeout:     -100 * time.Millisecond,
+			retries:     1000,
+			expectedErr: fmt.Sprintf("Timeout in Healthcheck cannot be less than %s", container.MinimumDuration),
+		},
+		{
+			doc:         "test invalid Retries in Healthcheck: less than 0",
+			interval:    time.Second,
+			timeout:     time.Second,
+			retries:     -10,
+			expectedErr: "Retries in Healthcheck cannot be negative",
+		},
+		{
+			doc:         "test invalid StartPeriod in Healthcheck: not 0 and less than 1ms",
+			interval:    time.Second,
+			timeout:     time.Second,
+			retries:     1000,
+			startPeriod: 100 * time.Microsecond,
+			expectedErr: fmt.Sprintf("StartPeriod in Healthcheck cannot be less than %s", container.MinimumDuration),
+		},
+	}
+
+	for i, tc := range testCases {
```

Comment:
Can you use `t.Run()`?

Reply:
ha! I was considering suggesting that 🤗

Reply:
Thank you for the reviews. I have updated the PR.